proposals/dedht.md

```diff
@@ -0,0 +1,130 @@
+Good project scope aims for ~3-5 engineers for 1-3 months (though feel free to suggest larger-scoped projects anyway). 
+Projects do not include regular day-to-day maintenance and improvement work, e.g. on testing, tooling, validation, code clarity, refactors for future capability, etc.
+-->
+<!--
+For ease of discussion in PRs, consider breaking lines after every sentence or long phrase.
+-->
+
+## Purpose &amp; impact 
+#### Background &amp; intent
+_Describe the desired state of the world after this project? Why does that matter?_
+
+<!--
+Outline the status quo, including any relevant context on the problem you’re seeing that this project should solve. Wherever possible, include pains or problems that you’ve seen users experience to help motivate why solving this problem works towards top-line objectives. 
+-->
+
+**Discovery of content providers on the Filecoin network without centralized indexes.**
+
+There is currently an unsolved problem of knowing which miner is storing which content. If you have a CID, which miner do you retrieve the content from?
+
+Maintaining a global index of which miner has which content will be difficult (in terms of keeping up to date, it's size, the running costs etc.) and is inherently centralized.
+
+This project proposes that miners keep an index of the content that they are storing themselves (including of deals made up of batches/aggregations of content) such that they can be asked if content for a CID can be retrieved from them.
+
+Miners also register themselves on the "DeDHT", a DHT where deals are made with miners that are "closest" to the CID of the content. There is no guarantee that deals are actually made with CIDs that are "close" to the miner's ID but there is incentive to do so if the creator wants the content to be found efficiently by others.
+
+Leveraging a DHT avoids the need for a centralized index of all the content and allows miners to only have to keep track of the things they have agreed to store.
+
+Note that a file is made up of many CIDs. The recommendation could be that whole files reside on a single miner, it is just the root CID that is "close" to the miner. The same can apply for directories.
+
+One issue is aggregations, which will be slower to accumulate because content will need to be aggregated in multiple "buckets" of "close" CIDs.
+
+On the surface Sybil attacks appear to be a problem. The more places in the DHT your miners are in the more deals they will receive. However if we mandate that participants in this DHT must be active miners registered in the Filecoin blockchain the barrier to entry to becoming a miner should be deterrent enough to prevent this.
+
+#### Assumptions &amp; hypotheses
+_What must be true for this project to matter?_
+<!--(bullet list)-->
+
+* Users want to discover which Filecoin miner is storing their data.
+* We don't really want to keep a central index of where the CIDs are stored.
+
```

Comment:
If I understood the approach correctly, we are also assuming that either the deal price does not vary significantly across miners or the deal price is not a problem for the user.